### PR TITLE
support create, alter, drop, select view statement

### LINKED ISSUE #7138: Unsupported SQL of `CREATE VIEW`
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?

5.0.0-RC1-SNAPSHOT

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-Proxy

### Expected behavior

Support SQL:

`create view view1 as select * from t_order where id!=1;`

### Actual behavior

```
line 1:7 no viable alternative at input 'createview'
line 1:7 no viable alternative at input 'createview'
line 1:7 no viable alternative at input 'createview'
line 1:7 no viable alternative at input 'createview'
[ERROR] 22:20:05.284 [ShardingSphere-Command-7] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
org.apache.shardingsphere.sql.parser.exception.SQLParsingException: Unsupported SQL of `create view view1 as select * from t_order where id!=1`
	at org.apache.shardingsphere.rdl.parser.engine.executor.RDLSQLParserExecutor.execute(RDLSQLParserExecutor.java:55)
	at org.apache.shardingsphere.rdl.parser.engine.engine.RDLSQLParserEngine.parse0(RDLSQLParserEngine.java:70)
	at org.apache.shardingsphere.rdl.parser.engine.engine.RDLSQLParserEngine.parse(RDLSQLParserEngine.java:52)
	at org.apache.shardingsphere.rdl.parser.engine.ShardingSphereSQLParserEngine.parse(ShardingSphereSQLParserEngine.java:47)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:75)
	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.<init>(MySQLComQueryPacketExecutor.java:71)
	at org.apache.shardingsphere.proxy.frontend.mysql.command.MySQLCommandExecutorFactory.newInstance(MySQLCommandExecutorFactory.java:70)
	at org.apache.shardingsphere.proxy.frontend.mysql.command.MySQLCommandExecuteEngine.getCommandExecutor(MySQLCommandExecuteEngine.java:60)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:94)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:73)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

### PR DESCRIPTION
**Pull request by strongduanmu**
Fixes #7138.

Changes proposed in this pull request:
- support create, alter, drop statement
- add create, alter view validator
- add view_type in SchemaMetaDataLoader to support select view

### CODE REVIEW

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/util/StatementValidatorUtil.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import org.apache.shardingsphere.sql.parser.sql.common.segment.generic.table.SimpleTableSegment;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Statement validator utility class.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class StatementValidatorUtil {
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/util/StatementValidatorUtil.java:35]: Is this class just for some of the unsupported DDLSQLStataments?  In that case, do you think an abstract class to carry these functions will be better?
`StatementValidatorUtil` gives me a  feeling that it is born for all the statements.

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ShardingStatementValidatorFactory.java [hunk 2]
```diff
@@ -66,6 +73,25 @@
         if (sqlStatement instanceof CreateProcedureStatement) {
             return Optional.of(new ShardingCreateProcedureStatementValidator());
         }
+        if (sqlStatement instanceof CreateViewStatement) {
+            return Optional.of(new ShardingCreateViewStatementValidator());
+        }
+        if (sqlStatement instanceof AlterViewStatement) {
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ShardingStatementValidatorFactory.java:79]: If `Create view` SQL can work well for `single table` currently, do you think `ShardingUnconfiguredTablesRoutingEngine` will be a better match?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/util/StatementValidatorUtil.java (hunk 1 above)
Comment (strongduanmu) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/util/StatementValidatorUtil.java:35]: > Is this class just for some of the unsupported DDLSQLStataments? In that case, do you think an abstract class to carry these functions will be better?
> `StatementValidatorUtil` gives me a feeling that it is born for all the statements.

@tristaZero I agree with your point, how about adding a new abstract class `ShardingDDLStatementValidator` to carry these functions？

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ShardingStatementValidatorFactory.java (hunk 2 above)
Comment (strongduanmu) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ShardingStatementValidatorFactory.java:79]: > If `Create view` SQL can work well for `single table` currently, do you think `ShardingUnconfiguredTablesRoutingEngine` will be a better match?

@tristaZero `ShardingUnconfiguredTablesRoutingEngine` can meet the needs of creating views on a single data source. I think it is a better choice. 👍

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/dml/ShardingDMLStatementValidator.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import org.apache.shardingsphere.infra.binder.statement.SQLStatementContext;
+import org.apache.shardingsphere.infra.binder.type.TableAvailable;
+import org.apache.shardingsphere.infra.exception.ShardingSphereException;
+import org.apache.shardingsphere.sharding.route.engine.validator.ShardingStatementValidator;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.SQLStatement;
+
+/**
+ * Sharding dml statement validator.
+ */
+public abstract class ShardingDMLStatementValidator<T extends SQLStatement> implements ShardingStatementValidator<T> {
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/dml/ShardingDMLStatementValidator.java:29]: Oh, you did it!

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/model/schema/model/schema/SchemaMetaDataLoader.java [hunk 4]
```diff
@@ -42,6 +42,8 @@
     
     private static final String TABLE_TYPE = "TABLE";
     
+    private static final String VIEW_TYPE = "VIEW";
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/model/schema/model/schema/SchemaMetaDataLoader.java:45]: Why do we need `VIEW_TYPE`?

**Review by tristaZero — APPROVED** [review]
@strongduanmu Beautiful programming, organized and clear!

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/model/schema/model/schema/SchemaMetaDataLoader.java (hunk 4 above)
Comment (strongduanmu) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/model/schema/model/schema/SchemaMetaDataLoader.java:45]: > Why do we need `VIEW_TYPE`?

@tristaZero In order to support select view statement.

**Pull request merged**